### PR TITLE
Luxor/reduce council budget

### PR DESCRIPTION
**Pull request by dobertRowneySr**
addresses: #3494

Budget is simply reduced by the amount (there is no token minting until workers/councilors gets paid)
I have also added a proposal, with the same parameters as the `freeze_pallet` proposal
- [x] add extrinsic to council pallet
- [x] add tests to council pallet
- [x] add proposal to codex pallet
- [x] add tests to codex pallet
- [x] add runtime configuration
- [x] add integration tests

### CODE REVIEW

**Comment by kdembler** [comment]
@ignazio-bovo please change target branch to `luxor`

File: runtime-modules/proposals/codex/src/lib.rs
Comment (kdembler) [inline runtime-modules/proposals/codex/src/lib.rs]:
```suggestion
        /// Decrease Council budget parameters
```

File: runtime-modules/proposals/codex/src/weights.rs
Comment (kdembler) [inline runtime-modules/proposals/codex/src/weights.rs:1025]: This needs to be adjusted right?

File: runtime/src/proposals_configuration/defaults.rs
Comment (kdembler) [inline runtime/src/proposals_configuration/defaults.rs]: We need to adjust those, will get back to you

**Review by kdembler — COMMENTED** [review]
Look good to me, will approve once we finalize the params for proposal, waiting for council feedback

**Review by freakstatic — APPROVED** [review]
LGTM!

File: runtime/src/proposals_configuration/defaults.rs
Comment (kdembler) [reply runtime/src/proposals_configuration/defaults.rs]: Council has approved a signal to make this 3/3 approval, 1 constitutionality: https://pioneer.joyutils.org/#/proposals/preview/797

File: runtime-modules/proposals/codex/src/weights.rs
Comment (dobertRowneySr) [reply runtime-modules/proposals/codex/src/weights.rs:1025]: Hi, yes but they will be generated by @mnaamani on the ref. hardware

Comment (dobertRowneySr) [reply runtime-modules/proposals/codex/src/weights.rs:1025]: And not on this pr

File: runtime/src/proposals_configuration/defaults.rs
Comment (dobertRowneySr) [reply runtime/src/proposals_configuration/defaults.rs]: Let me know if also the quorum percentage is correct

File: textureAtlas.png
Comment (kdembler) [inline textureAtlas.png:1]: You probably didn't want to include this :D

**Review by mnaamani — CHANGES_REQUESTED** [review]
Runtime implementation looks good.
I think you ran the query-node build before the runtime spec was updated, that is why the query-node/chain-metadata/2002.json file is modified. Please revert it to one from master.
The integration test fails because the mappings do not handle case of the new proposal, so the processor fails to handle the event. Do you want to add the mapping and schema update for QN in this PR or a separately?

**Comment by dobertRowneySr** [comment]
The QN part is handled separately

File: runtime-modules/council/src/lib.rs
Comment (mnaamani) [inline runtime-modules/council/src/lib.rs]: Should we add a weight function for this dispatch

File: runtime-modules/proposals/codex/src/tests/mock.rs
Comment (mnaamani) [inline runtime-modules/proposals/codex/src/tests/mock.rs:107]: Why did you find it necessary to move the Council pallet down here.

**Review by mnaamani — CHANGES_REQUESTED** [review]
Just need to add a weight function for the `decrease_council_budget` dispatch call.

File: runtime-modules/council/src/lib.rs
Comment (dobertRowneySr) [reply runtime-modules/council/src/lib.rs]: This was part of a bigger issue that there are no benchmarks.
I have added that for the `reduce_council_budget`, thanks for pointing it out

File: runtime-modules/proposals/codex/src/tests/mock.rs
Comment (dobertRowneySr) [reply runtime-modules/proposals/codex/src/tests/mock.rs:107]:
```
error[E0277]: the trait bound `RuntimeEvent: From<pallet_council::RawEvent<u64, u64, u64, u64>>` is not satisfied
   --> runtime-modules/proposals/codex/src/tests/mock.rs:696:25
    |
696 |     type RuntimeEvent = RuntimeEvent;
    |                         ^^^^^^^^^^^^ the trait `From<pallet_council::RawEvent<u64, u64, u64, u64>>` is not implemented for `RuntimeEvent`
```

Comment (dobertRowneySr) [reply runtime-modules/proposals/codex/src/tests/mock.rs:107]: without that

File: query-node/mappings/src/proposals.ts
Comment (mnaamani) [inline query-node/mappings/src/proposals.ts:111]: I think this change is okay, although we probably should implement this function in future to account for any removed proposals between runtimes. To date we have only been adding new proposals.

File: runtime-modules/council/src/lib.rs
Comment (mnaamani) [inline runtime-modules/council/src/lib.rs:864]: when adding a new dispatch that is invoked by a proposal, it is best to try to add it at the end so that the method index of existing dispatches doesn't change, so we can have the option to not remove all pending proposals on runtime upgrades.

File: runtime-modules/proposals/codex/src/tests/mock.rs
Comment (mnaamani) [reply runtime-modules/proposals/codex/src/tests/mock.rs:107]: Is is something about the way this particular mock is implemented with respect to types. We do not seem to need to do the same for the main runtime/lib.rs

**Review by mnaamani — APPROVED** [review]
I think this is good to merge, just left some general comments.

**Pull request merged**
